routes/index.js

```diff
@@ -110,4 +108,94 @@ router.get(`/${APP_VERSION}/datastore_search`, async function (req, res, next) {
   }
 })
 
+//TODO finish this test function to manually check downloads
+// router.get(`/test/download`, async function (req, res, next) {
+//   // res.sendFile('./test-download.html')
+//   const ppath = __dirname.split(path.sep).slice(0, -1).join(path.sep)
+//   res.sendFile(path.join(ppath + '/test/test-download.html'))
+// })
+/**
+ *
+ */
+
+router.post(`/${APP_VERSION}/download`, async function (req, res, next) {
+  console.log(' Download CALLED')
+  // get the graphql query from body
+  const query = req.body.query ? req.body.query : req.body
+  // call GraphQL
+  try {
+    // TODO check graphql syntax BEFORE sending it
+    const gqlRes = await request(`${process.env.HASURA_URL}/v1/graphql`, query)
+
+    // // capture graphql response
+    const ext = (req.params.format || req.query.format || 'json')
+      .toLowerCase()
+      .trim()
+    const colSep = (req.query.field_separator || ',').trim()
+    res.set(
+      'Content-Disposition',
+      'attachment; filename="download.' + ext + '";'
+    )
+    if (ext != 'json') {
```

Comment:
this is why I'm saying xls is practically default - _if user sends anything except 'json' (e.g. 'pdf') as the `ext` param_ this condition resolves to true, so the app will return data in xls (at least that's what I'm seeing in the code, do we have a test case to prove me wrong?)

Reply:
@leomrocha ^

Reply:
@EvgeniiaVak Now I understand what you meant. 

Line 131 gives the default as json when nothing is passed. In the code if the format is not supported by the js-xlsx extension it would anyways return an error as the library wouldn't be able to handle it. If the format is supported and not json, it should be by definition a tabular format, so that's why the condition is done that way. It doesn't default to excel, it goes to whatever tabular format was given by the client, the only check before is for CSV as the separator has special treatment. Any other tabular format (supported by js-xlsx) will then be used, this includes for example ODS and older excel formats.

Maybe this deserves a more in depth discussion on what should be the default behaviour if the format is not supported.
1. Should we return an error? -> I would vote for this, returning then a 400 Bad Request, the check would be after setting the format in line 131, if the format is not supported. I will start adding this.
2. Should we return json? -> In this case, instead of changing the `if` conditions I would constrain the line 131 (below) to just accept valid types or default back to json:
```javascript
const ext = (req.params.format || req.query.format || 'json')
      .toLowerCase()
      .trim()
```